Track project events

iref opened the pull request:
# What's in this PR?

Segment plug tracker now also tracks events for project ids if id is extractable from request resource.
There is more work to be done. Mostly to add `acceptor` property in `Approved Project Membership` event. This change might require passing of optional current user to `SegmentTraitBuilder.build`

## References

Closes #1285
Progress on: #927

iref committed:
Add function to extract project id from resources

iref committed:
Add event tracking for projects to segment plug

iref commented:
@joshsmith please let me know if these changes match what we discussed in #1307 .

joshsmith reviewed (commented):
This is generally moving in the right direction, though refer back to this comment I had about the event names needing to change to be explicit about the directionality:

> This means we'd actually rewrite the events as `Membership Requested (Project)` and `Approved Membership (Project)` from the perspective of the project. But we would also track a `Requested Membership (User)` and `Membership Approved (User)` from the perspective of the user.

iref committed:
Add event direction to tracking event names

iref committed:
Move get_project_id to preserve alphabetic ordering.

iref committed:
Add acceptor to project user events

Tracking of project user events was moved to project_user controller.
This change allows us to use existing tracking api without need of
passing additional arguments like current user or action to
SegmentTraitsBuilder.build/2 that arent needed in other explicitly
tracked events.

iref committed:
Rename current_user in update event to acceptor

iref committed:
Removed unnecessary project_id? function

iref committed:
Remove unnecessary id param in event name builder

begedin reviewed (commented):
Looking at this further, I'm not so sure we should be tracking an event identifying as a project. 

The way I see it, from the project's POV, we should still track which user (project admin or owner), approved the membership, or invited a user. 

We know which project it happened on from the traits themselves. If we suddenly start recording events triggered by users which are actually project records, our segment dashboard will become confusing.

@joshsmith What do you think?
